File: python/two_sums.py
import typing


def two_sums(nums: typing.List[int], target: int) -> typing.Tuple[int, int]:
    visited = {}
    
    for index, element in enumerate(nums):
        sub_target = abs(target - element)
        if sub_target in visited:
            return index, visited[sub_target]
        else:
            visited[element] = index


if __name__ == '__main__':
    print(two_sums([3, 4, 7, 8, 4, 8], 10))

File: go/two_sums.go
package main

import (
	"errors"
	"fmt"
	"math"
)

func two_sums(nums []int, target int) ([2]int, error) {
	visited := map[int]int{}

	for index, element := range nums {
		sub_target := int(math.Abs(float64(target - element)))
		_, ok := visited[sub_target]
		if ok {
			return [2]int{index, visited[sub_target]}, nil
		} else {
			visited[element] = index
		}
	}
	return [2]int{0, 0}, errors.New("Failed")
}

func main() {
	arr := [...]int{3, 4, 7, 9, 0, 3}
	fmt.Println(two_sums(arr[:], 10))
}
